cmd/main.go

```diff
@@ -53,7 +58,7 @@ func main() {
 	case "argocd-k8s-auth":
 		command = k8sauth.NewCommand()
 	default:
-		command = cli.NewCommand()
+		command = cli.NewDefaultArgoCDCommandWithArgs(o)
 	}
 
 	if err := command.Execute(); err != nil {
```

Comment:
As discussed offline between @leoluz, @nitishfy and @christianh814 we can use the error path here to fallback to the plugin lookup, which would alleviate the performance concerns stated above.

Reply:
@blakepettersson @nitishfy In this case we just need to make sure that the error returned by Cobra is in fact a "command not found". Unfortunately Cobra doesn't return an specific error type for that case. This PR implements it but is still open by the time of this writing: https://github.com/spf13/cobra/pull/2167

We shouldn't trigger the plugin lookup if the command execution returns anything different than "unknown command..". Pattern matching on the error message isn't the direction I would prefer to go :/

Reply:
@leoluz damn, I thought something so obvious would have already been implemented 😞

Reply:
@leoluz need your review now. I am yet to remove the dead code [here](https://github.com/argoproj/argo-cd/pull/20074/files#diff-36169dc3aa841cc1f8978f1a720ae31fd95846331ecc28170be4e6e89ea62542R39), it is only getting used for the test, but I can fix that too. Let me know what do you think about the current approach.

Reply:
@nitishfy I think this is much better. Unfortunately it still doesn't address the issue that I am referring here in this thread. I explained the error scenario a bit better in the comment below.

Reply:
As mentioned before you need to ensure that only "unknown command" errors would be handled by this flow

```suggestion
	if err := command.Execute(); err != nil && strings.Contains(err.Error(), "unknown command") {
```

In an ideal world this would need to be an actual type as @leoluz mentioned, but for now for this to work as intended this would need to do. We need to work on getting the actual Cobra PR merged so that we can negate this hacky way of doing it.